src/TBuffer.cpp

```diff
@@ -4431,3 +4439,68 @@ void TBuffer::encodingChanged(const QString& newEncoding)
         }
     }
 }
+
+QString TBuffer::processSupportsRequest(const QString& elements)
+{
+    // strip initial SUPPORT and tokenize all of the requested elements
+    auto elementsList = elements.trimmed().remove(0, 7).split(QStringLiteral(" "), QString::SkipEmptyParts);
+    QStringList result;
+
+    auto reportEntireElement = [](auto element, auto& result) {
+        result.append("+" + element);
+
+        for (const auto& attribute : mSupportedMxpElements.value(element)) {
+            result.append("+" + element + QStringLiteral(".") + attribute);
+        }
+
+        return result;
+    };
+
+    auto reportAllElements = [reportEntireElement](auto& result) {
+        auto elementsIterator = mSupportedMxpElements.constBegin();
+        while (elementsIterator != mSupportedMxpElements.constEnd()) {
+
+            // make this a lambda to append element + all attributes
+            result = reportEntireElement(elementsIterator.key(), result);
+        }
+
+        return result;
+    };
+
+    // empty <SUPPORT> - report all known elements
+    if (elementsList.isEmpty()) {
+        result = reportAllElements(result);
+    } else {
+        // otherwise it's <SUPPORT element1 element2 element3>
+        for (auto& element : elementsList) {
+            // prune any enclosing quotes
+            if (element.startsWith(QChar('"'))) {
+                element = element.remove(0, 1);
+            }
+            if (element.endsWith(QChar('"'))) {
+                element.chop(1);
+            }
+
+            if (!element.contains(QChar('.'))) {
+                result.append((mSupportedMxpElements.contains(element) ? "+" : "-") + element);
+            } else {
+                auto elementName = element.section(QStringLiteral("."), 0, 0);
+                auto attributeName = element.section(QStringLiteral("."), 1, 1);
+
+                if (!mSupportedMxpElements.contains(elementName)) {
```

Comment:
The specification is unclear in this place: If the server requests `<SUPPORT "color.*">` and we don't support `color` at all, should the client send `-color` or `-color.fore`?

Thinking of it, the former is probably right as the client doesn't know anything about the tag if it's not supported and thus can't correctly send any supported attributes back.

Reply:
Yep.